File: week1_baseline/python/src/boukensha/run_dsl.py
from __future__ import annotations

from typing import Callable

from .tool import Tool


class RunDSL:
    def __init__(self, registry) -> None:
        self._registry = registry

    def tool(self, name: str, *, description: str, parameters: dict | None = None, block: Callable) -> Tool:
        return self._registry.tool(name, description=description, parameters=parameters, block=block)

File: week2_capable/dotnet/src/Boukensha.Core/RunDsl.cs
namespace Boukensha.Core;

public sealed class RunDsl(Registry registry)
{
    public ToolDefinition Tool(
        string name,
        string description,
        IReadOnlyDictionary<string, ToolParameter>? parameters,
        Func<IReadOnlyDictionary<string, object?>, Task<string>> handler) =>
        registry.Tool(name, description, parameters, handler);
}
